pkg/collector/corechecks/containers/containerd.go

```diff
@@ -0,0 +1,381 @@
+// ContainerdFactory is used to create register the check and initialize it.
+func ContainerdFactory() check.Check {
+	return &ContainerdCheck{
+		CheckBase: corechecks.NewCheckBase(containerdCheckName),
+		instance:  &ContainerdConfig{},
+		sub:       &Subscriber{},
+	}
+}
+
+// Parse is used to get the configuration set by the user
+func (co *ContainerdConfig) Parse(data []byte) error {
+	if err := yaml.Unmarshal(data, co); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Configure parses the check configuration and init the check
+func (c *ContainerdCheck) Configure(config, initConfig integration.Data) error {
+	var err error
+	if err = c.CommonConfigure(config); err != nil {
+		return err
+	}
+
+	c.hostname, err = util.GetHostname()
+	if err != nil {
+		return err
+	}
+
+	if err = c.instance.Parse(config); err != nil {
+		return err
+	}
+	c.sub.Filters = c.instance.Filters
+	return nil
+}
+
+// Run executes the check
+func (c *ContainerdCheck) Run() error {
+	sender, err := aggregator.GetSender(c.ID())
+	defer sender.Commit()
+	if err != nil {
+		return err
+	}
+	// As we do not rely on a singleton, we ensure connectivity every check run.
+	cu, errHealth := cutil.GetContainerdUtil()
+	if errHealth != nil {
+		log.Infof("Error ensuring connectivity with Containerd daemon %v", errHealth)
+		return errHealth
+	}
+	ns := cu.Namespace()
+
+	if c.sub == nil {
+		c.sub = CreateEventSubscriber("ContainerdCheck", ns, c.instance.Filters)
+	}
+
+	if !c.sub.IsRunning() {
+		// Keep track of the health of the Containerd socket
+		c.sub.CheckEvents(cu)
+	}
+	events := c.sub.Flush(time.Now().Unix())
+	// Process events
+	computeEvents(c.hostname, events, sender, c.instance.Tags)
+
+	nk := namespaces.WithNamespace(context.Background(), ns)
+	computeMetrics(sender, nk, cu, c.instance.Tags)
+
+	return nil
+}
+
+// compute events converts Containerd events into Datadog events
+func computeEvents(hostname string, events []ContainerdEvent, sender aggregator.Sender, userTags []string) {
+	for _, e := range events {
+		split := strings.Split(e.Topic, "/")
+		if len(split) != 3 {
+			// sanity checking the event, to avoid submitting
+			log.Tracef("Event topic %s does not have the expected format", e.Topic)
+			continue
+		}
+
+		output := metrics.Event{
+			Priority:       metrics.EventPriorityNormal,
+			Host:           hostname,
+			SourceTypeName: containerdCheckName,
+			EventType:      containerdCheckName,
+			AggregationKey: fmt.Sprintf("containerd:%s", e.Topic),
+		}
+		output.Text = e.Message
+		if len(e.Extra) > 0 {
+			for k, v := range e.Extra {
+				output.Tags = append(output.Tags, fmt.Sprintf("%s:%s", k, v))
+			}
+		}
+		output.Tags = append(output.Tags, userTags...)
+		output.Ts = e.Timestamp.Unix()
+		output.Title = fmt.Sprintf("Event on %s from Containerd", split[1])
+		if split[1] == "containers" || split[1] == "tasks" {
+			// For task events, we use the container ID in order to query the Tagger's API
+			tags, err := tagger.Tag(e.ID, true)
```

Comment:
I was able to confirm that the containerids match:

from the node:
```
~$ sudo /usr/local/bin/ctr -n k8s.io c ls
CONTAINER                                                           IMAGE                                                                                                RUNTIME
c34bc1f0dafd17a346011fc61391c67c48664e9ff1594df8c298c1e37c81c30d    727006795293.dkr.ecr.us-east-1.amazonaws.com/datadog-agent:20181219-internal-6.8.2-jmx               io.containerd.runtime.v1.linux
```

From the agent:
```
=== Entity containerd://c34bc1f0dafd17a346011fc61391c67c48664e9ff1594df8c298c1e37c81c30d ===
Tags: [app:datadog-agent chart_version:0.10.17 container_id:c34bc1f0dafd17a346011fc61391c67c48664e9ff1594df8c298c1e37c81c30d display_container_name:trace-agent_datadog-agent-w8x7z image_name:727006795293.dkr.ecr.us-east-1.amazonaws.com/datadog-agent image_tag:20181219-internal-6.8.2-jmx kube_container_name:trace-agent kube_daemon_set:datadog-agent kube_namespace:datadog-agent pod_name:datadog-agent-w8x7z service:datadog-agent short_image:datadog-agent]
Sources: [kube-metadata-collector kubelet]
```